src/AirplaneSecondary.java

```diff
@@ -0,0 +1,37 @@
+/**
+ * The secondary methods for AirplaneKernel.
+ */
+public interface AirplaneSecondary {
```

Comment:
This would be your abstract class:
```suggestion
public abstract class AirplaneSecondary implements Airplane {
```